storm-server/src/main/java/org/apache/storm/metricstore/NimbusMetricProcessor.java

```diff
@@ -24,7 +24,7 @@
     public void processWorkerMetrics(Map<String, Object> conf, WorkerMetrics metrics) throws MetricException {
         try (NimbusClient client = NimbusClient.getConfiguredClient(conf)) {
             client.getClient().processWorkerMetrics(metrics);
-        } catch (TException e) {
```

Comment:
Could we instead of handling all exceptions, be more specific on capturing `NimbusLeaderNotFoundException` ?
`catch (TException | NimbusLeaderNotFoundException e)`

Reply:
done